### PR TITLE
*: use macros from slog rather than Rust standard log

### PR DESCRIPTION
**Pull request by Hijiao**
Signed-off-by: Max <xiahaijiao@hotmail.com>

## What have you changed? (mandatory)
This is the first step of the RFC  ["unified log format"](https://github.com/tikv/rfcs/blob/log-format-standard/text/2018-12-19-unified-log-format.md).

In this PR, macros like `debug!()`,`info!()`, `error!()` are imported from `slog` rather than Rust standard `log`. 

## What are the type of the changes? (mandatory)
- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)
No

## Does this PR affect documentation (docs) update? (mandatory)
No

### CODE REVIEW

**Comment by Hijiao** [comment]
@overvenus PTAL, thanks.

**Comment by breeswish** [comment]
What about

```
#[macro_use]
extern crate slog_scope;
```

Additionally, we are not libraries, so it's mostly fine with slog-scope (still, explicitly passing the logger is better, but we can do it later anyway).

File: components/test_raftstore/Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,12 @@ futures = "0.1"
 tokio-timer = "0.2"
 grpcio = { version = "0.4", features = [ "secure" ] }
 rand = "0.3"
-log = "0.3.9"
+log = { version = "0.3", features = ["release_max_level_debug"] }
```
Comment (breeswish) [inline components/test_raftstore/Cargo.toml]: we might not really need this change in test_xxx

File: components/test_raftstore/src/cluster.rs [hunk 2]
```diff
@@ -192,18 +192,18 @@ impl<T: Simulator> Cluster<T> {
     }
 
     pub fn run_node(&mut self, node_id: u64) {
-        debug!("starting node {}", node_id);
+        debug!(::slog_scope::logger(), "starting node {}", node_id);
```
Comment (siddontang) [inline components/test_raftstore/src/cluster.rs]: IMO, I don't like this way, too verbose, weird

File: src/util/mod.rs [hunk 3]
```diff
@@ -498,20 +497,8 @@ pub fn set_panic_hook(panic_abort: bool, data_dir: &str) {
...
-                drainer,
-                logger::convert_log_level_to_slog_level(level),
-                false, // Use sync logger to avoid an unnecessary log thread.
-                false, // It is initialized already.
-            ) {
-                g.cancel_reset();
-            }
-        }
+        // To collect remaining logs, drop the guard before exit.
+        ::slog_global::clear_global();
```
[the remark this answers is not in the document]
Comment (breeswish) [reply src/util/mod.rs]: Yes. Tested manually using a simplified case.

[the remark this answers is not in the document]
Comment (overvenus) [reply src/util/mod.rs]: Why not outputs to the stderr? there may be some valuable log.

[the remark this answers is not in the document]
Comment (breeswish) [reply src/util/mod.rs]: You mean output further logs to stderr?

[the remark this answers is not in the document]
Comment (overvenus) [reply src/util/mod.rs]: Yes. I dont see any changes on slog-global

**Comment by breeswish** [comment]
We need to test:
1. Whether stderr output works
2. Whether trace is filtered out at compile time

So marking as DNM first.

**Pull request merged**